### PR TITLE
Log statistic used to roll initiative in Combatant flags

### PR DESCRIPTION
**Pull request by stwlam**

### CODE REVIEW

File: src/module/actor/base.ts [hunk 1]
```diff
@@ -710,8 +713,12 @@ class ActorPF2e extends Actor<TokenDocumentPF2e, ItemTypeMap> {
 
         const initiativeRoll = Math.trunc(participant.initiative);
         rollOptionsAll[`self:participant:initiative:roll:${initiativeRoll}`] = true;
-        const rank = [...participants].reverse().indexOf(participant) + 1;
+        const rank = participants.indexOf(participant) + 1;
         rollOptionsAll[`self:participant:initiative:rank:${rank}`] = true;
+        if (participant.flags.pf2e.initiativeStatistic) {
+            const stat = participant.flags.pf2e.initiativeStatistic;
```
Comment (simonward) [inline src/module/actor/base.ts:719]: Is it reasonable to pull this out of the if block so it can be used in the if test itself?

Comment (stwlam) [reply src/module/actor/base.ts:719]: I don't follow

Comment (simonward) [reply src/module/actor/base.ts:719]: `const stat = participant.flags.pf2e.initiativeStatistic;
if (stat) {`

Comment (stwlam) [reply src/module/actor/base.ts:719]: There's no sense in setting a roll option of `"self:participant:initiative:stat:null": false`.

Comment (simonward) [reply src/module/actor/base.ts:719]: `if (stat) {`
Should address that.
This is more about the cost of accessing the properties twice, which is probably negligible anyway.

Comment (CarlosFdez) [reply src/module/actor/base.ts:719]: Its a very minor change to strip out that doesn't save any lines. It'd be a better idea if this was javascript but in typescript its not a big deal.

File: src/module/encounter/document.ts [hunk 2]
```diff
@@ -92,8 +95,18 @@ class EncounterPF2e extends Combat {
...
-            result ? { id: result.combatant.id, value: result.roll.total } : []
+        const initiatives = rollResults.flatMap((result): SetInitiativeData | never[] =>
+            result
+                ? {
+                      id: result.combatant.id,
+                      value: result.roll.total,
+                      statistic:
+                          result.roll.options.domains?.find(
+                              (s): s is SkillLongForm | "perception" =>
+                                  setHasElement(SKILL_LONG_FORMS, s) || s === "perception"
```
Comment (CarlosFdez) [inline src/module/encounter/document.ts:106]: Is it possible to roll initiative with lore skills? If so, I don't know if this is a safe way to find the rolled skill.

Comment (stwlam) [reply src/module/encounter/document.ts:106]: The general rule is that lore skills can only be used for recall knowledge, but it's a special ability away from overriding that (e.g. Exploit Vulnerability--haven't surveyed for any allowing one to use a lore skill for initiative). Implementation-wise, the system only offers non-lore skills as options.

Comment (CarlosFdez) [reply src/module/encounter/document.ts:106]: blargh

**Pull request merged**